### ISSUE TITLE
Improve errors system

### ISSUE THREAD

**Issue by povilasjurcys** [samesystem/graphql_rails#49]
It's possible to customize errors for each project by extending `GraphqlRails::Error` class, but it's not very intuitive and it's not documented. We need to do both: improve errors API and document it

**Comment by kraftingas** [comment]
It would be nice to know how exactly errors should be handled from UI side. This could be implemented by having `code` next to each error. Here is one of the examples taken from [here](https://medium.com/@takewakamma/graphql-error-handling-with-graphql-ruby-653aa2a129f6):

```
  "errors": [
    {
      "message": "permission denied",
      "path": [
        "createUser"
      ],
      "extensions": {
        "code": "AUTHENTICATION_ERROR"
      }
    }
  ]
```

[Here](https://graphql-ruby.org/errors/execution_errors.html#customizing-error-json) is a documentation how this could be implemented.

**Comment by povilasjurcys** [comment]
fixed by #124
